src/words/ast/INodeQueueWait.java

```diff
@@ -10,7 +17,33 @@ public INodeQueueWait(Object... children) {
 
 	@Override
 	public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+		ASTValue referenceObject = children.get(0).eval(environment);
+		ASTValue identifier = children.get(1).eval(environment);
+		AST lengthExpression = children.get(2);
+		ASTValue doNow = children.get(3) != null ? children.get(3).eval(environment) : null;
+
+		WordsObject object;
+		if (referenceObject.type.equals(ASTValue.ValueType.OBJ)){
+			WordsProperty property = referenceObject.objValue.getProperty(identifier.stringValue);
+			if (property.type != WordsProperty.PropertyType.OBJECT) {
+				throw new WordsInvalidTypeException(ASTValue.ValueType.OBJ.toString(), property.type.toString());
+			}
+			object = property.objProperty;
+		} else {
+			object = environment.getObject(identifier.stringValue);
+		}
+		if (object == null) {
```

Comment:
This pattern of code from lines 25:34 appears in several places.  I'm going to open an issue to factor this out as a separate method.